content/slice/language-guide/result-types.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Result types
+description: Learn how to define and use results in Slice.
+---
+
+{% slice1 %}
+The built-in generic type Result<Success, Failure> is not available with Slice1.
+{% /slice1 %}
+
+{% slice2 %}
+
+## Result
+
+A result is a built-in generic type that holds either a Success value or a Failure value, where `Success` and `Failure`
+are type parameters.
+
+You construct a result inline, without giving it a name, usually to specify the return type of an operation. For
+example:
+
+```slice
+module VisitorCenter
+
+interface Greeter {
+    greet(name: string) -> Result<string, GreeterError>
+}
+
+/// The errors returned by the Greeter.
+unchecked enum GreeterError {
+    EmptyName
+    NameTooLong(maxLength: int32)
+    Away(until: WellKnownTypes::TimeStamp)
+}
+```
+
+You can also use a constructed result as the type of a field, as the element type in a sequence, as the type of an
```

Comment:
`of` is correct here, `in` would be for the actual elements themselves, not the their types.
Also, all the redundant `as`s sound strange to me.
```suggestion
You can also use a constructed result as the type of a field, the element type of a sequence, the type of an
```

Reply:
Fixed